tests/Data/ExtendAndImplementsTest.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mvorisek\Atk4\Hintable\Tests\Data;
+
+use Atk4\Core\Phpunit\TestCase;
+use Atk4\Data\Model;
+use Atk4\Data\Persistence;
+use Mvorisek\Atk4\Hintable\Tests\Data\Model\IModelAddonFields;
+use Mvorisek\Atk4\Hintable\Tests\Data\Model\ModelExtendsAndImplements;
+
+class ExtendAndImplementsTest extends TestCase
+{
+    protected function createPersistence(): Persistence
+    {
+        $db = new \Atk4\Data\Persistence\Array_();
+        $db->onHook(Persistence::HOOK_AFTER_ADD, function (Persistence $p, Model $m) {
+            if (is_a($m, IModelAddonFields::class, true)) {
+                $m->addField('field_in_interface');
+            }
+        });
+
+        return $db;
+    }
+
+    public function testRefBasic(): void
```

Comment:
"Ref" kw refers to refecences, this does not test refence is any way. It tests annotated field.

Also, any annotations from interfaces should have lower priority than annotations from classes (as based on the definition/inheritance nesting?).

Reply:
Why priority if a field name is unique?

Reply:
unique yes, but if declared wrongly (or covariant)...